### PR TITLE
Add filter-level security test coverage; remove dead PenetrationTests scaffolding

### PR DESCRIPTION
**Pull request by bonnieshi43**
Add filter-level security test coverage; remove dead PenetrationTests scaffolding

Add 65 new unit tests (6 files) covering the AbstractSecurityFilter subclasses that had zero coverage: DefaultAuthorizationFilter, AnonymousUserFilter, InvalidateSessionFilter, LogoutFilter, and SecurityHeaderFilter, plus a cross-filter ordering test that chains multiple real filter instances through the existing FilterTestSupport harness to verify the implicit ordering contracts between them (StandardFilterChain / SecurityFilterChain). All tests are tagged @Tag("core") and run as part of the normal test suite.

- DefaultAuthorizationFilterTest (13 cases): unauthorized/redirect gate, redirect URL construction, org-cookie cleanup, fresh-anonymous-session tracking, and the EM permission if/else-if branch.
- AnonymousUserFilterTest (11 cases): exempt paths, successful anonymous auth establishing a session, session-limit failure handling (401 JSON / forward), and the /vs-events silent no-op path.
- InvalidateSessionFilterTest (15 cases): stale-session invalidation, the logout/sessionexpired exemption, security-disabled bypass, and curr_org_id cleanup on portal requests.
- LogoutFilterTest (11 cases): /logout and /sessionexpired routing, and the open-redirect / no-CSRF-protection scenarios below.
- SecurityHeaderFilterTest (9 cases): X-Frame-Options, X-XSS-Protection, X-Content-Type-Options, X-Robots-Tag branches.
- SecurityFilterChainOrderingTest (6 cases): DefaultAuthorizationFilter vs CSRFFilter evaluation order, InvalidateSessionFilter handoff, and AnonymousUserFilter/DefaultAuthorizationFilter ordering.

Two confirmed defects were found while writing these tests and are tracked as passing characterization tests (current behavior) plus @Disabled regression-guard tests (correct post-fix behavior, ready to
enable once each fix lands):

- Bug #75656: DefaultAuthorizationFilter.java:73 compares  principal.getName() with equals(XPrincipal.ANONYMOUS), but
  SRPrincipal.getName() always returns the full identity key  (name + delimiter + org), never the bare literal, so an  established  anonymous principal is never recognized as anonymous on subsequent  requests --  ontainsAnonymous(orgID) is silently skipped. 
  Confirmed  via SecurityFilterChainOrderingTest to be a full authorization bypass  (not just a wasted session) when AnonymousUserFilter runs before  DefaultAuthorizationFilter.
- Bug #75658: AbstractLogoutFilter.getLogoutRedirectUri() reflects an   unvalidated "redirectUri" request parameter into the post-logout  redirect (open redirect) when fromEm=true and the request is a GET;
  combined with LogoutFilter requiring no CSRF token and running ahead  of CSRFFilter in the chain, this is triggerable via a simple /<a>  tag against an authenticated victim.

Remove the old annotation-driven endpoint-coverage scaffolding (PenetrationTests, PenetrationTest, PenetrationTestsExtension,PenetrationCoverageTest): @PenetrationTest had zero usages anywhere in
the codebase, and PenetrationCoverageTest itself is tagged @Tag("penetration") while core/pom.xml's surefire config only runs <groups>core</groups>, so it never actually executed in CI. It is fully superseded by the FilterTestSupport-based tests above. Verified no other references exist anywhere in the repo before deleting.

### CODE REVIEW

File: core/src/test/java/inetsoft/web/security/InvalidateSessionFilterTest.java [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+
+      filter.doFilter(request, response, chain);
+
+      verify(chain).doFilter(request, response);
+      // Safe despite setUp() stubbing mockEngine.getSecurityProvider() / mockProvider
+      // .getAuthenticationProvider() via when(...): Mockito excludes invocations made purely to
+      // set up a stub from interaction verification (verifyNoInteractions/verifyNoMoreInteractions
+      // only count invocations made by the code under test). Confirmed empirically -- this test
+      // passes both in isolation and as part of the full class.
+      verifyNoInteractions(mockEngine, mockProvider);
```
[the remark this answers is not in the document]
Comment (bonnieshi43) [reply core/src/test/java/inetsoft/web/security/InvalidateSessionFilterTest.java:115]: mvn test -Dtest=InvalidateSessionFilterTest 没有问题. 可以忽略

**Review by jshobe-inetsoft — APPROVED** [review]
Claude's follow-up review (after commits dd8ba7df, bad0395c) confirms all prior nits were fixed with real, verified changes. The one open question (Mockito verifyNoInteractions semantics in InvalidateSessionFilterTest) is resolved by the green 'build' check, which includes this test. No blocking issues remain. Approving.
